### PR TITLE
Look for libdevice in additional paths.

### PR DESCRIPTION
**Pull request by maleadt**

### CODE REVIEW

**Comment by Maaarcocr** [comment]
```julia
function find_libdevice(targets::Vector{VersionNumber}, toolkit_dirs)
    @debug("Request to look $(source_str(toolkit_dirs)) for libdevice $(join(targets, ", "))")

    # figure out locations
    dirs = String[]
    for toolkit_dir in toolkit_dirs
        push!(dirs, toolkit_dir)
        push!(dirs, joinpath(toolkit_dir, "libdevice"))
        push!(dirs, joinpath(toolkit_dir, "nvvm", "libdevice"))
    end

    # filter
    dirs = valid_dirs(dirs)
    @debug("Looking $(source_str(dirs)) for libdevice")
    isempty(dirs) && return nothing

    # select
    for dir in dirs 

    # parse filenames
        libraries = Dict{VersionNumber,String}()
        for target in targets
            path = joinpath(dir, "libdevice.compute_$(target.major)$(target.minor).10.bc")
            if isfile(path)
                libraries[target] = path
            end
        end
        library = nothing
        let path = joinpath(dir, "libdevice.10.bc")
            if isfile(path)
                @debug("Found libdevice at $dir")
                library = path
            end
        end

        # select
        if library != nothing
            @debug("Found unified libdevice at $library")
            return library
        elseif !isempty(libraries)
            @debug("Found libdevice for ", join(sort(map(ver->"sm_$(ver.major)$(ver.minor)",
                                            keys(libraries))), ", ", " and "), " at $dir")
            return libraries
        end
    end
    return nothing
end
```

This seems to work. What I added is that I loop through the directories in dirs, so that we check in each of them. It seemed to me that this was not the case before.

**Comment by nsmith5** [comment]
Will this work if libdevice is hiding at `/usr/share/cuda/libdevice.10.bc`? If not I'll file an issue for that as `find_libdevice` is failing for me under those conditions.

File: src/discovery.jl [hunk 1]
```diff
@@ -238,6 +238,7 @@ function find_toolkit()
         end
         append!(dirs, basedirs)
         push!(dirs, "/usr/lib/nvidia-cuda-toolkit")
+        push!(dirs, "/usr/share/cuda")
```
Comment (nsmith5) [inline src/discovery.jl:241]: Yes! Thank you for this. Not sure if you're working under Fedora with the negativo17 repo, but this has been burning me all day.

**Comment by maleadt** [comment]
> Will this work if libdevice is hiding at /usr/share/cuda/libdevice.10.bc? If not I'll file an issue for that as find_libdevice is failing for me under those conditions.

It should. `/usr/share/cuda` is added to the toolkit dirs, and being looked at as part of:
https://github.com/JuliaGPU/CUDAapi.jl/pull/37/files#diff-a14992c78e0523f8b1bdfb65ea99cdfcR281

Did you test this PR? I'm not using Fedora, so I don't have a easy way to test this.

File: src/discovery.jl (hunk 1 above)
Comment (Maaarcocr) [reply src/discovery.jl:241]: I'm working under Fedora and brought it up on the slack and @maleadt immediately solved the issue 😄

**Comment by nsmith5** [comment]
Can test. I'll get back to you as soon as possible.

**Comment by nsmith5** [comment]
Yup, works now.

```julia
julia> using CUDAapi

julia> dirs = find_toolkit()
3-element Array{String,1}:
 "/usr"           
 "/"              
 "/usr/share/cuda"

julia> find_libdevice([v"9.1"], dirs)
"/usr/share/cuda/libdevice.10.bc"

```

Unfortunately, `find_host_compiler()` fails. I'll file a separate bug for this (its called `cuda-gcc` when using the repositories I'm using...).

**Comment by maleadt** [comment]
Great, thanks for testing.

**Pull request merged**